library/request_queue.cpp

```diff
@@ -90,33 +79,44 @@ dnet_io_req *dnet_request_queue::pop_request(dnet_work_io *wio, const char *thre
 	FORMATTED(HANDY_TIMER_STOP, ("pool.%s.queue.wait_time", thread_stat_id), (uint64_t)r);
 
 	auto cmd = static_cast<dnet_cmd *>(r->header);
-	const auto expired = [&r, &cmd, this] () {
-		if (!m_timeout || (cmd->flags & DNET_FLAGS_NO_QUEUE_TIMEOUT))
+	auto st = r->st;
+	auto node = st->n;
+	const auto timeout = [&node, &cmd]() -> uint64_t {
```

Comment:
Fixed: since lambda is called immediately there is no need to restrict captured objects.